src/mpid/ch4/netmod/ofi/ofi_am_impl.h

```diff
@@ -315,9 +316,18 @@ static inline int MPIDI_OFI_am_isend_short(int rank,
     iov[2].iov_base = (void *) data;
     iov[2].iov_len = data_sz;
 
+    MPI_Aint total_len = sizeof(*msg_hdr) + am_hdr_sz + data_sz;
+    if (total_len & 0x7) {
+        /* pad the data to 8-byte alignment */
+        static char pad[16];
+        iov[3].iov_base = pad;
+        iov[3].iov_len = 8 - (total_len & 0x7);
+        num_iov = 4;
```

Comment:
I thought we used to check if the provider `iov_limit` was >= 3 to ensure support for the active message APIs. However, I can't find it, so maybe it is only an implicit assumption. The bump to 4 probably doesn't make any difference.

Reply:
@raffenet In our discussion for GPUs, we wanted to move away from the IOV model for AMs.  It's confusing and is an artifact of what CH3 used to have.  The intent here is to send three pointers: (1) user buffer, (2) header, and (3) optional extended header.  I think those three should be explicit parameters to the function, rather than sticking them inside an IOV.  Just pointing it out in case you are considering adding even more content to the IOV.

Reply:
@pavanbalaji Any idea how to address the alignment issue when the payload is not multiple of required alignment? I currently can think of two solutions:
* Don't use `FI_MULTI_RECV` -- need evaluate and determine how much benefit of `FI_MULTI_RECV` bring us.
* Stage the payload buffer -- the buffer need come form persistent allocations and the additional memcpy will hurt large payload.

Reply:
Note that the suggestion for libfabric to fix the alignment issue has been turned down -- https://github.com/ofiwg/libfabric/issues/6041

Reply:
We could copy the header out before using it.  We did the same thing for `ch3` too.

Reply:
> 
> 
> We could copy the header out before using it. We did the same thing for `ch3` too.

OK, I'll try that.